src/relay.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  Address,
+  BaseError,
+  createPublicClient,
+  createWalletClient,
+  ContractFunctionRevertedError,
+  getContract,
+  http,
+  WalletClient,
+} from "viem";
+
+import { celo, celoAlfajores } from "viem/chains";
+import config from "./config";
+import getSecret from "./get-secret";
+import { relayerAbi } from "./relayer-abi";
+import { privateKeyToAccount } from "viem/accounts";
+
+function getPublicClient() {
+  const isMainnet = process.env.NODE_ENV !== "development";
+  return createPublicClient({
+    chain: isMainnet ? celo : celoAlfajores,
+    transport: http(),
+  });
+}
+
+async function getWalletClient(): Promise<WalletClient> {
+  const isMainnet = process.env.NODE_ENV !== "development";
+  const pk = await getSecret(config.RELAYER_PK_SECRET_ID);
+
+  return createWalletClient({
+    account: privateKeyToAccount(pk as Address),
+    chain: isMainnet ? celo : celoAlfajores,
+    transport: http(),
+  });
+}
+
+export default async function relay(relayerAddress: string): Promise<boolean> {
+  const publicClient = getPublicClient();
```

Comment:
given this function is gonna be called pretty often, we should already do some performance optimization here and move the `getPublicClient()` and `getWalletClient()` calls outside the function if possible. Should initialize these clients once and reuse them across function invocations.

https://cloud.google.com/functions/docs/bestpractices/tips#use_global_variables_to_reuse_objects_in_future_invocations

Reply:
I'm not sure if we can still do this given that the wallet client initialization is async. Created https://github.com/mento-protocol/oracle-relayer/issues/6 to look into this later

Reply:
yeah for async stuff fair point, need to investigate, left a comment in #6 for a possible approach